test/test_helper.rb

```diff
@@ -7,4 +7,18 @@ class ActiveSupport::TestCase
   fixtures :all
 
   # Add more helper methods to be used by all tests here...
+  def login_user(user)
+    session[:user_id] = user.id
+  end
+
+  OmniAuth.config.test_mode = true
+  OmniAuth.config.mock_auth[:github] = OmniAuth::AuthHash.new({
+        :provider => 'github',
+        :info => {
+          :nickname => 'angela4713'
+        },
+        :credentials => {
+          :token => SecureRandom.hex(8)
+        }
+      })
```

Comment:
I suggest you wrap this in a method so that the mock returns data corresponding to the user that's getting authenticated. So that the changes on this file look like this:
```ruby
OmniAuth.config.test_mode = true

 # Add more helper methods to be used by all tests here...
def login_user(user)
  authenticate_with_github(user)
  session[:user_id] = user.id
end

def authenticate_with_github(user)
  OmniAuth.config.mock_auth[:github] = OmniAuth::AuthHash.new({
    provider: 'github',
    info: {
      nickname: user. username
    },
    credentials: {
      token: user.token
    }
   })
end
```